distribution/lib/Standard/Table/0.0.0-dev/src/Data/Type/Enso_Types.enso

```diff
@@ -22,13 +23,16 @@ most_specific_value_type value use_smallest=False =
         _ : Date        -> Value_Type.Date
         _ : Time_Of_Day -> Value_Type.Time
         _ : Date_Time   -> Value_Type.Date_Time
-        i : Integer     -> case use_smallest of
-            False -> Value_Type.Integer Bits.Bits_64
-            True ->
-                storage_type = IntegerType.smallestFitting i
-                value_type = Storage.to_value_type storage_type
-                # We do a small rewrite here - for integers we always return the Integer type, even if the value is small enough to fit in a Byte.
-                if value_type == Value_Type.Byte then Value_Type.Integer Bits.Bits_16 else value_type
+        i : Integer     ->
+            case NumericConverter.isBigInteger i of
+                False -> case use_smallest of
+                    False -> Value_Type.Integer Bits.Bits_64
+                    True ->
+                        storage_type = IntegerType.smallestFitting i
+                        value_type = Storage.to_value_type storage_type
+                        # We do a small rewrite here - for integers we always return the Integer type, even if the value is small enough to fit in a Byte.
+                        if value_type == Value_Type.Byte then Value_Type.Integer Bits.Bits_16 else value_type
+                True -> Value_Type.Decimal precision=Nothing scale=0
```

Comment:
Ah, now I understand better why `Value_Type.Decimal` is being used for `BigInteger`.